2023/10/19/main.go

```diff
@@ -0,0 +1,33 @@
+// Daily Joel 2023-10-19
+//
+// Variable declaration and initialization, variable scopes,
+// and variadic parameters.
+package main
+
+import (
+	"fmt"
+)
+
+func concatenate(s ...string) string {
+	out := ""
+	for _, ss := range s {
+		out += ss
+	}
+	return out
+}
+
+var a, b string = "N", "o"
+var c string = "e"
+var d string
+
+func printJoel() {
+	a := "J"
+	d = "l"
+	fmt.Println(concatenate(a, b, c, d))
+}
```

Comment:
Since a is assigned within the function it will be prioritised over the previous assignment of a

Reply:
Exactly. Now, there are two variables named `a` and the one declared inside `printJoel()` is considered inside that function